HighAvailabilityModule.Storage.Client/SQLStorageMembershipClient.cs

```diff
@@ -0,0 +1,392 @@
+            }
+            else
+            {
+                throw new FormatException("Input value is not string[].");
+            }
+        }
+
+        public async Task<byte[]> TryGetByteArray(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string valueTmp = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.Byte[]")
+            {
+                string[] s = valueTmp.Split(",");
+                byte[] value = new byte[s.Length];
+                for (int i=0; i<s.Length; i++)
+                {
+                    value[i] = byte.Parse(s[i]);
+                }
+                return value;
+            }
+            else
+            {
+                throw new FormatException("Input value is not byte[].");
+            }
+        }
+
+        public async Task SetDataEntryAsync(string path, string key, string value, string type)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string lastSeenValue = getDataEntry.Item1;
+            string lastSeenType = getDataEntry.Item2;
+
+            SqlConnection con = new SqlConnection(this.ConStr);
+            string StoredProcedure = SetDataEntrySpName;
+            SqlCommand comStr = new SqlCommand(StoredProcedure, con);
+            comStr.CommandType = CommandType.StoredProcedure;
+            comStr.CommandTimeout = Convert.ToInt32(Math.Ceiling(this.OperationTimeout.TotalSeconds));
+
+            comStr.Parameters.Add("@dpath", SqlDbType.NVarChar).Value = path;
+            comStr.Parameters.Add("@dkey", SqlDbType.NVarChar).Value = key;
+            comStr.Parameters.Add("@dvalue", SqlDbType.NVarChar).Value = value;
+            comStr.Parameters.Add("@dtype", SqlDbType.NVarChar).Value = type;
+            comStr.Parameters.Add("@lastSeenValue", SqlDbType.NVarChar).Value = lastSeenValue;
+            comStr.Parameters.Add("@lastSeenType", SqlDbType.NVarChar).Value = lastSeenType;
+
+            try
+            {
+                await con.OpenAsync();
+                await comStr.ExecuteNonQueryAsync();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error occured when setting data entry: {ex.ToString()}");
+                throw new InvalidOperationException($"Error occured when setting data entry: {ex.ToString()}");
+            }
+            finally
+            {
+                con.Close();
+                con.Dispose();
+                comStr.Dispose();
+            }
+        }
+
+        public async Task SetGuid(string path, string key, Guid value)
+        {
+            await SetDataEntryAsync(path, key, value.ToString(), value.GetType().ToString());
+        }
+
+        public async Task SetString(string path, string key, string value)
+        {
+            await SetDataEntryAsync(path, key, value.ToString(), value.GetType().ToString());
```

Comment:
`value` is already `string`? Also we already know the type of giving value here. We should avoid the reflection cost imposed by `GetType`